astronomer/providers/amazon/aws/example_dags/example_sagemaker.py

```diff
@@ -0,0 +1,158 @@
+    bucket_name = f"amazon-sagemaker-example-{str(uuid.uuid4())}"
+    ecr_repository_name = f"{ACCOUNT_ID}.dkr.ecr.us-east-2.amazonaws.com/providers_repo"
+    processing_job_name = f"{str(uuid.uuid4())}"
+    processing_local_input_path = "/opt/ml/processing/input"
+    processing_local_output_path = "/opt/ml/processing/output"
+    raw_data_s3_key_input = "preprocessing/input/validation.csv"
+    clean_data = "preprocessing/output/clean_data.csv"
+
+    ecr_repository_image_uri = (
+        f"{ACCOUNT_ID}.dkr.ecr.us-east-2.amazonaws.com/providers_repo:sagemaker-processing"
+    )
+    resource_config = {
+        "InstanceCount": 1,
+        "InstanceType": "ml.m5.large",
+        "VolumeSizeInGB": 1,
+    }
+    processing_config = {
+        "ProcessingJobName": processing_job_name,
+        "ProcessingInputs": [
+            {
+                "InputName": "input",
+                "AppManaged": False,
+                "S3Input": {
+                    "S3Uri": f"s3://{bucket_name}/{raw_data_s3_key_input}",
+                    "LocalPath": processing_local_input_path,
+                    "S3DataType": "S3Prefix",
+                    "S3InputMode": "File",
+                    "S3DataDistributionType": "FullyReplicated",
+                    "S3CompressionType": "None",
+                },
+            },
+        ],
+        "ProcessingOutputConfig": {
+            "Outputs": [
+                {
+                    "OutputName": "output",
+                    "S3Output": {
+                        "S3Uri": f"s3://{bucket_name}/{clean_data}",
+                        "LocalPath": processing_local_output_path,
+                        "S3UploadMode": "EndOfJob",
+                    },
+                    "AppManaged": False,
+                }
+            ]
+        },
+        "ProcessingResources": {"ClusterConfig": resource_config},
+        "StoppingCondition": {"MaxRuntimeInSeconds": 3000},
+        "AppSpecification": {
+            "ImageUri": ecr_repository_image_uri,
+        },
+        "RoleArn": role_arn,
+    }
+
+    ti = get_current_context()["ti"]
+    ti.xcom_push(key="bucket_name", value=bucket_name)
+    ti.xcom_push(key="raw_data_s3_key_input", value=raw_data_s3_key_input)
+    ti.xcom_push(key="ecr_repository_name", value=ecr_repository_name)
+    ti.xcom_push(key="processing_config", value=processing_config)
+
+
+with DAG(
+    dag_id="example_sagemaker_rajath",
+    start_date=datetime(2021, 8, 13),
+    schedule_interval=None,
+    catchup=False,
+    tags=["example", "sagemaker", "async", "AWS"],
+) as dag:
+
+    setup_aws_config = BashOperator(
+        task_id="setup_aws_config",
+        bash_command=f"aws configure set aws_access_key_id {AWS_ACCESS_KEY_ID}; "
+        f"aws configure set aws_secret_access_key {AWS_SECRET_ACCESS_KEY}; "
+        f"aws configure set default.region {AWS_DEFAULT_REGION}; ",
+    )
+
+    test_setup = set_up(
```

Comment:
why is this required?

Reply:
in this method, we have set up the details for sagemaker processing job config and downstream tasks to pick from xcom of this. 
Also more details would be added as part of training and transform which is in other PR needed for downstream tasks